### ISSUE TITLE
Add HPO integration tests

### ISSUE THREAD

**Issue by zeinali72** [zeinali72/swe-pinn#98]
## Context

The HPO pipeline has **zero automated tests**. Changes to \`src/training/\`, \`src/losses/\`, or \`src/models/\` can silently break the HPO without anyone knowing until a multi-hour study fails. This is especially critical after the train loop rewrite (#91).

## Related Issues

- **Should follow**: #91 (train loop rewrite — test the new code)
- **Related**: #62 (zero unit tests — broader test coverage initiative)
- **Parallel-safe** with: #92, #93, #94, #95, #96, #97

## Proposed Change

Add \`test/test_hpo.py\` with minimal integration tests:

\`\`\`python
"""Integration tests for the HPO pipeline."""
import os
os.environ["JAX_PLATFORM_NAME"] = "cpu"

import unittest
import optuna
from optimisation.objective_function import objective, get_hpo_value

class TestGetHpoValue(unittest.TestCase):
    """Test the config-driven parameter suggestion helper."""
    
    def test_fixed_scalar_returns_value(self):
        trial = optuna.trial.create_trial(values=[0.5])
        # Fixed scalar should return as-is
        result = get_hpo_value(trial, "test_param", 42.0)
        self.assertEqual(result, 42.0)

    def test_list_returns_categorical(self):
        # List should trigger suggest_categorical
        ...

    def test_dict_range_returns_float(self):
        # Dict with min/max should trigger suggest_float
        ...

class TestObjectiveSmoke(unittest.TestCase):
    """Smoke test: run a tiny HPO trial end-to-end."""
    
    def test_objective_returns_valid_float(self):
        \"\"\"Run objective() with 2 epochs on a tiny model.\"\"\"
        base_config = {
            "scenario": "experiment_1",
            "training": {"epochs": 2, "learning_rate": 1e-3, "batch_size": 32, "seed": 42},
            "model": {"name": "MLP", "width": 16, "depth": 2, "output_dim": 3},
            "domain": {"lx": 100.0, "ly": 10.0, "t_final": 60.0},
            "physics": {"g": 9.81, "n_manning": 0.03, "u_const": 0.29, "inflow": None},
            "sampling": {"n_points_pde": 64, "n_points_ic": 32, "n_points_bc_domain": 32},
            "loss_weights": {"pde_weight": 1.0, "ic_weight": 1.0, "bc_weight": 1.0, "neg_h_weight": 1.0, "data_weight": 0.0},
            "device": {"dtype": "float32"},
            "numerics": {"eps": 1e-6},
            "validation_grid": {"n_points_val": 64},
            "hpo_hyperparameters": {},
        }
        
        study = optuna.create_study(direction="maximize")
        study.optimize(lambda trial: objective(trial, base_config), n_trials=1)
        
        self.assertEqual(len(study.trials), 1)
        result = study.trials[0].value
        self.assertIsInstance(result, float)
        # Should be a valid number (not NaN, not -1.0 for a trivial 2-epoch run)
\`\`\`

## Files to Create

- \`test/test_hpo.py\`

## Files to Read (for test design)

- \`test/test_train.py\` — existing test patterns (CPU forcing, mock data, teardown)
- \`test/test_assets/test_config.yaml\` — minimal config template

## Acceptance Criteria

- [ ] \`python -m unittest test.test_hpo\` passes
- [ ] Tests run on CPU (\`JAX_PLATFORM_NAME=cpu\`)
- [ ] Tests complete in <30 seconds (tiny model, 2 epochs)
- [ ] At least: \`get_hpo_value\` unit tests + 1 end-to-end smoke test
